components/setup/BuildInstruction.tsx

```diff
@@ -0,0 +1,110 @@
+}
+
+const BuildInstruction: React.FC<BuildInstructionProps> = ({ selectedHosting, userId, selectedProvider, selectedInstallationType, session }) => {
+	const [isButtonDisabled, setIsButtonDisabled] = useState<boolean>(false);
+	const [buildStatus, setBuildStatus] = useState<CloudBuildStatus | null>(null);
+	const [isRepoSelectionDone, setIsRepoSelectionDone] = useState<boolean>(false);
+	const [installId, setInstallId] = useState<string | null>(null);
+	const [isGetInstallIdLoading, setIsGetInstallIdLoading] = useState(false);
+
+	const handleBuildButtonClick = () => {
+		setIsButtonDisabled(true);
+		setBuildStatus(null);
+
+		axios.post('/api/dpu/trigger', { userId })
+			.then((response) => {
+				console.log('[handleBuildButtonClick] /api/dpu/trigger response:', response.data);
+				setBuildStatus(response.data);
+				if (response.data.success) {
+					return;
+				}
+			})
+			.catch((error) => {
+				console.error('[handleBuildButtonClick] /api/dpu/trigger request failed:', error);
+				setIsButtonDisabled(false);
+				setBuildStatus({ success: false, message: 'API request failed' });
+			})
+	};
+
+	useEffect(() => {
+		setIsGetInstallIdLoading(true);
+		axios.post('/api/dpu/pubsub', { userId }).then(async (response) => {
+			if (response.data.installId) {
+				setInstallId(response.data.installId);
+			}
+		})
+			.catch((error) => {
+				console.error(`[BuildInstruction] Unable to get topic name for user ${userId} - ${error.message}`);
+			})
+			.finally(() => {
+				setIsGetInstallIdLoading(false);
+			});
+	}, [userId])
+
+	if (selectedHosting === 'selfhosting') {
+		if (isGetInstallIdLoading) {
+			return (<>
+				<div className='inline-block border-4 border-t-primary-main rounded-full w-6 h-6 animate-spin mx-2'></div>
+				Generating topic name...
+			</>);
+		}
+		if (!installId) {
+			return (<div className="flex items-center gap-4">
+				<p>Something went wrong while fetching install id.</p>
+			</div>);
+		}
+		if (!isRepoSelectionDone && (
+			(selectedProvider === 'bitbucket' && selectedInstallationType === 'project') || 
+			(selectedProvider === 'github' && selectedInstallationType === 'individual')
+		)) {
+			return (<RepoSelection repoProvider={selectedProvider} installId={installId} setIsRepoSelectionDone={setIsRepoSelectionDone} />)
+		}
+		return <DockerInstructions userId={userId} selectedInstallationType={selectedInstallationType} selectedProvider={selectedProvider} session={session} installId={installId} />
+	} else if (selectedHosting === 'cloud') {
+		if (selectedInstallationType === 'project') {
+			return (
+				<div className="flex items-center gap-4">
+					<Button variant="contained" onClick={handleBuildButtonClick} disabled={isButtonDisabled}>
+						Trigger Cloud Build
+					</Button>
+					<BuildStatus buildStatus={buildStatus} isButtonDisabled={isButtonDisabled} />
+				</div>
+			);
+		} else {
+			// TODO - if repo selection is false show repos, if true pat text field and trigger cloud build button
```

Comment:
This needs to be implemented before merging, right?

Reply:
At least the repo selection part, not the pat part necessarily

Reply:
Nahi nahi, it will all be part of Muskan's PR only. 
Showing RepoSelection component when there is no next step would not be good UX.